lighthouse-core/test/audits/valid-source-maps-test.js

```diff
@@ -0,0 +1,99 @@
+    const artifacts = {
+      URL: {finalUrl: 'https://example.com'},
+      ScriptElements: [],
+      SourceMaps: [],
+    };
+
+    const auditResult = await ValidSourceMaps.audit(artifacts);
+    expect(auditResult.score).toEqual(1);
+  });
+
+  it('passes when all large, first-party JS have corresponding source maps', async () => {
+    const artifacts = {
+      URL: {finalUrl: 'https://example.com'},
+      ScriptElements: [
+        {src: 'https://example.com/script1.min.js', content: largeBundle.content},
+        {src: 'https://example.com/script2.min.js', content: largeBundle.content},
+      ],
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/script1.min.js', map: largeBundle.map},
+        {scriptUrl: 'https://example.com/script2.min.js', map: largeBundle.map},
+      ],
+    };
+
+    const auditResult = await ValidSourceMaps.audit(artifacts);
+    expect(auditResult.score).toEqual(1);
+  });
+
+  it('fails when any large, first-party JS has no corresponding source map', async () => {
+    const artifacts = {
+      URL: {finalUrl: 'https://example.com'},
+      ScriptElements: [
+        {src: 'https://example.com/script1.min.js', content: largeBundle.content},
+        {src: 'https://example.com/script2.min.js', content: largeBundle.content},
+      ],
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/script1.min.js', map: largeBundle.map},
+        //  Missing corresponding source map for large, first-party JS (script2.min.js)
+      ],
+    };
+
+    const auditResult = await ValidSourceMaps.audit(artifacts);
+    expect(auditResult.score).toEqual(0);
+  });
+
+  it('passes when small, first-party JS have no corresponding source maps', async () => {
+    const artifacts = {
+      URL: {finalUrl: 'https://example.com'},
+      ScriptElements: [
+        {src: 'https://example.com/script1.min.js', content: largeBundle.content},
+        {src: 'https://example.com/script2.min.js', content: smallBundle.content},
+      ],
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/script1.min.js', map: largeBundle.map},
+        //  Missing corresponding source map for small, first-party JS (script2.min.js)
+      ],
+    };
+
+    const auditResult = await ValidSourceMaps.audit(artifacts);
+    expect(auditResult.score).toEqual(1);
+  });
+
+  it('passes when large, third-party JS have no corresponding source maps', async () => {
+    const artifacts = {
+      URL: {finalUrl: 'https://example.com'},
+      ScriptElements: [
+        {src: 'https://example.com/script1.min.js', content: largeBundle.content},
+        {src: 'https://d36mpcpuzc4ztk.cloudfront.net/script2.js', content: largeBundle.content},
+      ],
+      SourceMaps: [
+        {scriptUrl: 'https://example.com/script1.min.js', map: largeBundle.map},
+      ],
+    };
+
+    const auditResult = await ValidSourceMaps.audit(artifacts);
+    expect(auditResult.score).toEqual(1);
+  });
```

Comment:
can you add tests for:

1. missing source contents
1. values of subitems (can sprinkle throughout existing failure tests)

Reply:
where can I get source maps that have missing source contents? I can't find any rn

Reply:
you can just set values as `undefined` as needed. you'll need to load the map data within the `it` so as not to mutate for other test cases.

Reply:
Sounds good, added